### PR TITLE
MAYA-102168 clear payloads

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Add command to clear payloads.
- Make commands to clear refs and payloads be undoable.
- Improve Python options utility helpers functions to deal with dictionaries directly.
- Implement clear dialog UI in Python.
- Remember user choices between invocations.

### CODE REVIEW

**Review by AramAzhari-adsk — APPROVED** [review]

File: lib/mayaUsd/resources/scripts/mayaUsdLibRegisterStrings.py [hunk 1]
```diff
@@ -70,6 +70,16 @@ def mayaUsdLibRegisterStrings():
...
 
+    # mayaUsdClearRefsOrPayloadsOptions.py
+    register('kClearRefsOrPayloadsOptionsTitle', 'Clear All USD References/Payloads')
+    register('kClearRefsOrPayloadsOptionsMessage', 'Clear all references/payloads on %s?')
+    register('kClearButton', 'Clear')
+    register('kCancelButton', 'Cancel')
+    register('kAllRefsLabel', 'All References')
+    register('kAllRefsTooltip', 'Clear all references on the prim.')
+    register('kAllPayloadsLabel', 'All Payloads')
+    register('kAllPayloadsTooltip', 'Clearr all payloads on the prims.')
```
Comment (seando-adsk) [inline lib/mayaUsd/resources/scripts/mayaUsdLibRegisterStrings.py:81]: Typo in the string "Clearr" -> "Clear"

File: lib/mayaUsd/resources/scripts/mayaUsdClearRefsOrPayloadsOptions.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    """
+    Retrieves the current clear references or payloads options as a dictionary.
+    """
+    return mayaUsdOptions.getOptionsDict(
+        _getClearRefsOrPayloadsOptionsVarName(),
+        getDefaultClearRefsOrPayloadsOptionsDict())
+
+def setClearRefsOrPayloadsOptionsDict(options):
+    """
+    Sets the current clear references or payloads options from teh given dictionary.
```
Comment (seando-adsk) [inline lib/mayaUsd/resources/scripts/mayaUsdClearRefsOrPayloadsOptions.py:183]: Minor - typo in comment "teh" -> "the"

File: lib/mayaUsd/resources/scripts/mayaUsdClearRefsOrPayloadsOptions.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+
+def _createClearRefsOrPayloadsOptionsDialog():
+    """
+    Creates the clear references or payloads dialog UI.
+    """
+
+    # Same spacing as Maya option boxes.
+    spacer = 16
+    edge   = 12
```
Comment (seando-adsk) [inline lib/mayaUsd/resources/scripts/mayaUsdClearRefsOrPayloadsOptions.py:56]: Where did you get these values from? They are double the ones defined in mayaUsdMergeToUSDOptions.py

**Review by seando-adsk — CHANGES_REQUESTED** [review]
One minor typo in a UI string that should be fixed before merging.

File: lib/mayaUsd/resources/scripts/mayaUsdClearRefsOrPayloadsOptions.py (hunk 3 above)
Comment (pierrebai-adsk) [reply lib/mayaUsd/resources/scripts/mayaUsdClearRefsOrPayloadsOptions.py:56]: With the other values, the buttons looked too cramped compared to the design in the story.

Comment (pierrebai-adsk) [reply lib/mayaUsd/resources/scripts/mayaUsdClearRefsOrPayloadsOptions.py]: I re-tested it with the values from the other file and it does not look too bad, so I will use them as they are the Maya defaults.

**Review by seando-adsk — APPROVED** [review]
Looks good thanks.

**Review by AramAzhari-adsk — APPROVED** [review]

**Pull request merged**